internal/services/service.go

```diff
@@ -104,17 +140,76 @@ func Run(services []Service) {
 	case <-wait:
 		logger.Info.Println("main: all services were shutdown")
 	case <-time.After(DefaultServiceTimeout):
-		logger.Warning.Println("main: exited after timeout")
+		logger.Warning.Println("main: exited after a timeout")
+	}
+}
+
+func healthcheckPort(serviceName string) string {
+	if serviceName == "state-server" {
+		if address, ok := os.LookupEnv("SS_SERVER_ADDRESS"); ok {
+			split := strings.Split(address, ":")
+			if len(split) > 1 {
+				return split[1]
+			}
+		}
+		return DefaultStateServerPort
+	}
+
+	env := healthcheckEnv(serviceName)
+	if port, ok := os.LookupEnv(env); ok {
+		if serviceName == "state-server" {
+			split := strings.Split(port, ":")
+			return split[1]
+		}
+		return port
+	} else {
+		return DefaultHealthcheckPort
+	}
+}
+
+func healthcheckEnv(serviceName string) string {
+	suffix := "_HEALTHCHECK_PORT"
+	if serviceName == "dispatcher" || serviceName == "authority-claimer" {
+		suffix = "_HTTP_SERVER_PORT"
 	}
+	normalizedName := strings.Replace(serviceName, "-", "_", -1)
+	return fmt.Sprintf("%s%s", strings.ToUpper(normalizedName), suffix)
 }
 
 var (
-	GraphQLServer Service = simpleService{
-		serviceName: "graphql-server",
-		binaryName:  "cartesi-rollups-graphql-server",
+	AdvanceRunner = Service{
+		name:            "advance-runner",
+		binaryName:      "cartesi-rollups-advance-runner",
+		healthcheckPort: healthcheckPort("advance-runner"),
+	}
+	AuthorityClaimer = Service{
+		name:            "authority-claimer",
+		binaryName:      "cartesi-rollups-authority-claimer",
+		healthcheckPort: healthcheckPort("authority-claimer"),
+	}
+	Dispatcher = Service{
+		name:            "dispatcher",
+		binaryName:      "cartesi-rollups-dispatcher",
+		healthcheckPort: healthcheckPort("dispatcher"),
+	}
+	GraphQLServer Service = Service{
+		name:            "graphql-server",
+		binaryName:      "cartesi-rollups-graphql-server",
+		healthcheckPort: healthcheckPort("graphql-server"),
+	}
+	Indexer Service = Service{
+		name:            "indexer",
+		binaryName:      "cartesi-rollups-indexer",
+		healthcheckPort: healthcheckPort("indexer"),
+	}
+	InspectServer = Service{
+		name:            "inspect-server",
+		binaryName:      "cartesi-rollups-inspect-server",
+		healthcheckPort: healthcheckPort("inspect-server"),
 	}
-	Indexer Service = simpleService{
-		serviceName: "indexer",
-		binaryName:  "cartesi-rollups-indexer",
+	StateServer = Service{
+		name:            "state-server",
+		binaryName:      "cartesi-rollups-state-server",
+		healthcheckPort: healthcheckPort("state-server"),
 	}
 )
```

Comment:
I feel that this is in the wrong place. This pkg doesn't need to know about the instantiation of the node. Probably, this should be on the cmd or a `node` pkg.

Reply:
> This pkg doesn't need to know about the instantiation of the node

I agree. I'll put it in `validator.go` for now and after we get rid of Cobra I'll revisit this.

Reply:
Actually, I changed my mind. There's enough boilerplate code already to justify a `node` package.